apps/server/src/app/users/users.service.ts

```diff
@@ -57,11 +46,15 @@ export class UsersService {
    * @param: update dto and id of instance
    * @return: object with number of effected rows
    */
-  async updateUser(userParams: UpdateUser, userId: string) {
-    const affectedUsers = await this.UsersModel.update(userParams, {
-      where: { id: userId },
-    });
-    return affectedUsers;
+  async updateUser(userParams: User, userId: string) {
+    try {
+      const affectedUsers = await this.UsersModel.update(userParams, {
+        where: { id: userId },
+      });
+      return affectedUsers;
+    } catch (e) {
+      throw new DuplicateUsernameError();
```

Comment:
```suggestion
      throw new DuplicateUsernameError(params);
```
You should pass params to your errors and then print those params as part of the error message to be more informative.
For example: 
`// service.ts
throw new UserNotFoundError(userId);

// error.ts 
constructor(userId: string) {
message = `User ${this.userId} wasn't found`} `